### PR TITLE
op25 (new formula)

### PR DESCRIPTION
**Pull request by mikemorris**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/op25.rb [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+  depends_on "cmake" => :build
+  depends_on "cppunit" => :build
+  depends_on "swig" => :build
+  depends_on "gnuradio"
+  depends_on "gr-osmosdr"
+  depends_on "itpp"
+
+  def install
+    # TODO: apply patch if gnuradio >= 3.8?
```
Comment (mikemorris) [inline Formula/op25.rb:14]: Currently expects to build against `gnuradio@3.7` as explained in https://git.osmocom.org/op25/commit/?id=c26b51e0c1b5fd9eee0c5b13dc137050cd0c988b, refs https://github.com/Homebrew/homebrew-core/pull/49733

File: Formula/op25.rb [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+    args = std_cmake_args + %w[
+      -Wno-dev
```
Comment (mikemorris) [inline Formula/op25.rb:17]: Suppresses warnings intended for project development, but they're only visible when installing with `--verbose`, should this just be removed?

File: Formula/op25.rb [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    ENV.prepend_path "PYTHONPATH", libexec
+
+    # Needed because sys.path.append breaks when rx.py is called from symlink
+    ENV.prepend_path "PYTHONPATH", libexec/"tdma"
+    ENV.prepend_path "PYTHONPATH", libexec/"tx"
+
+    libexec.install Dir["#{buildpath}/op25/gr-op25_repeater/apps/**"]
+
+    # FIXME: is this an appropriate way to install?
+    bin.install_symlink libexec/"rx.py" => "op25"
```
Comment (mikemorris) [inline Formula/op25.rb:43]: I'm unsure if any of the Python application scripts in the `apps` directory should actually be renamed or installed to `bin` or if only installing into `libexec` would be more appropriate.

File: Formula/op25.rb [hunk 4]
```diff
@@ -0,0 +1,64 @@
+class Op25 < Formula
+  desc "Software-defined analyzer for APCO P25 signals"
+  homepage "https://osmocom.org/projects/op25/wiki"
+  url "git://op25.osmocom.org/op25.git"
+
+  depends_on "cmake" => :build
+  depends_on "cppunit" => :build
+  depends_on "swig" => :build
+  depends_on "gnuradio"
+  depends_on "gr-osmosdr"
```
Comment (mikemorris) [inline Formula/op25.rb:10]: Successfully builds against `gr-osmosdr@0.1.4`, will need to be updated if/when https://github.com/Homebrew/homebrew-core/pull/50921 is unblocked and merged.

File: Formula/op25.rb [hunk 5]
```diff
@@ -0,0 +1,64 @@
+class Op25 < Formula
+  desc "Software-defined analyzer for APCO P25 signals"
+  homepage "https://osmocom.org/projects/op25/wiki"
+  url "git://op25.osmocom.org/op25.git"
+
+  depends_on "cmake" => :build
+  depends_on "cppunit" => :build
+  depends_on "swig" => :build
+  depends_on "gnuradio"
```
Comment (mikemorris) [inline Formula/op25.rb:9]: I believe this defaults to building with macOS system Python 2.7 but I think should also build successfully with Python 3 (refs https://github.com/Homebrew/homebrew-core/pull/54848). I tested building against the `gnuradio@3.8` formula from https://github.com/Homebrew/homebrew-core/pull/49733 and it appeared to work.

File: Formula/op25.rb [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    system "make"
+    system "make", "install"
+
+    # Configure PYTHONPATH for runtime dependencies
+    [
+      Formula["gnuradio"].lib,
+      Formula["gr-osmosdr"].lib,
+      lib,
+    ].each do |dep|
+      Pathname.glob(dep/"python*/site-packages").each do |path|
```
Comment (mikemorris) [inline Formula/op25.rb:31]: Would using `Language::Python.major_minor_version` similar to https://github.com/Homebrew/homebrew-core/blob/master/Formula/urh.rb#L32 be a better option here?

**Comment by mikemorris** [comment]
`brew test-bot` seems to be stuck on a linker issue that I'm not hitting locally...
```
Undefined symbols for architecture x86_64:
  "pmt::dict_has_key(boost::intrusive_ptr<pmt::pmt_base> const&, boost::intrusive_ptr<pmt::pmt_base> const&)", referenced from:
      gr::basic_block::has_msg_port(boost::intrusive_ptr<pmt::pmt_base>) in fsk4_demod_ff_impl.cc.o
      gr::basic_block::has_msg_port(boost::intrusive_ptr<pmt::pmt_base>) in fsk4_slicer_fb_impl.cc.o
      gr::basic_block::has_msg_port(boost::intrusive_ptr<pmt::pmt_base>) in decoder_bf_impl.cc.o
      gr::basic_block::has_msg_port(boost::intrusive_ptr<pmt::pmt_base>) in decoder_ff_impl.cc.o
      gr::basic_block::has_msg_port(boost::intrusive_ptr<pmt::pmt_base>) in pcap_source_b_impl.cc.o
```

**Comment by SMillerDev** [comment]
@iMichka can you help here? It's python + gnuradio

**Comment by iMichka** [comment]
Our gnuradio uses system Python 2.

We can not accept new formulae that depend on a deprecated library.

#49733 is ready but has one issue; and I am still waiting from feedback from upstream to get it fixed, as I was not able to fix it myself. As long as #49733 is not merged, we can not accept the `op25` formula.

**Comment by mikemorris** [comment]
@iMichka Happy to wait for #49733 and update or reopen this targeting Python 3 and `gnuradio@3.8`

**Comment by SMillerDev** [comment]
I'd say reopening is the way to go here, given the uncertainty in the timing of https://github.com/Homebrew/homebrew-core/pull/49733

**Comment by mikemorris** [comment]
Missing/broken dependencies (for future reference):
- [ ] `zmq` and `waitress` Python libs for `-l http:localhost:PORT` web UI
- [ ] `GNUPLOT = /usr/bin/gnuplot` in `gr_gnuplot.py` possible to override with Python env wrapper to `GNUPLOT = /usr/local/bin/gnuplot`? Need to add `gnuplot` Hombrew dep too
- [ ] gnuplot `set terminal x11` broken on macOS, Homebrew-installed gnuplot appears to not support x11 terminal even with `brew cask install xquartz` - switching this to `set terminal qt` _works_ but appears to be hardcoded and not easily configurable

**Pull request closed without merging**